src/system-health/health_checker/service_checker.py

```diff
@@ -24,23 +43,171 @@ class ServiceChecker(HealthChecker):
 
     def __init__(self):
         HealthChecker.__init__(self)
+        self.critical_process_dict = {}
+        # Containers that has invalid critical_processes file
+        self.bad_containers = set()
+
+        self.container_feature_dict = {}
+
+        self.need_save_cache = False
+
+        self.load_critical_process_cache()
+
+    def get_expected_runnning_container_set(self, feature_table):
```

Comment:
Would it be better to rename the function name to `get_expected_running_containers(self, feature_table)`?